privacyidea/lib/resolvers/LDAPIdResolver.py

```diff
@@ -654,7 +654,7 @@ def testconnection(cls, param):
         
         return success, desc
 
-    def add_user(self, uid, object_class=None, attributes=None):
+    def add_user(self, uid, object_class=None, attributes=None, dn=None):
```

Comment:
OK, I see the problem now, since the uid (being the objectGUID or whatever is not available).

But this does not work with the normal REST API and the Web UI. I would like to keep this transparent for the toplevel and thus I would like to avoid to change the signature of this method.

What do you think of this.
To be able to keep this transparent to the levels above we need to create the DN out of the given attributes.

Usually you would give the loginname during creation of the user, the givenName and or the surname.

Then the DN could either be created like:

    CN=<givenname> <surname>, <baseDN>

or 

   CN=<loginname>, <baseDN>

To keep it simple, I would prefer to take the baseDN, which already exist. And then only define, what would be the creating attribute of the CN. (This would be an additional resolver parameter.)

Just tell me, what you think of this - I can do this later.

Till then, can you please remove the "dn" from the parameter list and... well ...at the moment just remove the add_user method? So we can merge the delete_user function, now.

Thanks a lot 
Cornelius